crates/isograph_compiler/src/source_files.rs

```diff
@@ -0,0 +1,370 @@
+            Err(BatchCompileError::MultipleErrors { messages: errors })
+        } else {
+            Ok(())
+        }
+    }
+
+    fn handle_update_schema(
+        &mut self,
+        config: &CompilerConfig,
+        event_kind: &SourceEventKind,
+    ) -> Result<(), BatchCompileError> {
+        match event_kind {
+            SourceEventKind::CreateOrModify(_) => {
+                self.schema = read_and_parse_graphql_schema(&config.schema)?;
+            }
+            SourceEventKind::Rename((_, target_path)) => {
+                if config.schema != *target_path {
+                    return Err(BatchCompileError::SchemaNotFound);
+                }
+            }
+            SourceEventKind::Remove(_) => return Err(BatchCompileError::SchemaNotFound),
+        }
+        Ok(())
+    }
+
+    fn handle_update_schema_extensions(
+        &mut self,
+        config: &CompilerConfig,
+        event_kind: &SourceEventKind,
+    ) -> Result<(), BatchCompileError> {
+        match event_kind {
+            SourceEventKind::CreateOrModify(path) => {
+                self.create_or_update_schema_extension(path)?;
+            }
+            SourceEventKind::Rename((source_path, target_path)) => {
+                if config.schema_extensions.contains(target_path) {
+                    self.create_or_update_schema_extension(target_path)?;
+                } else {
+                    let interned_file_path = intern_file_path(source_path);
+                    self.schema_extensions.remove(&interned_file_path);
+                }
+            }
+            SourceEventKind::Remove(path) => {
+                let interned_file_path = intern_file_path(path);
+                self.schema_extensions.remove(&interned_file_path);
+            }
+        }
+        Ok(())
+    }
+
+    fn handle_update_source_file(
+        &mut self,
+        config: &CompilerConfig,
+        event_kind: &SourceEventKind,
+    ) -> Result<(), BatchCompileError> {
+        match event_kind {
+            SourceEventKind::CreateOrModify(path) => {
+                self.create_or_update_iso_literals(&config.project_root, path)?;
+            }
+            SourceEventKind::Rename((source_path, target_path)) => {
+                let source_file_path = intern_file_path(source_path);
+                if self.contains_iso.remove(&source_file_path).is_some() {
+                    self.create_or_update_iso_literals(&config.project_root, target_path)?
+                }
+            }
+            SourceEventKind::Remove(path) => {
+                let interned_file_path = intern_file_path(path);
+                self.contains_iso.remove(&interned_file_path);
+            }
+        }
+        Ok(())
+    }
+
+    fn handle_update_source_folder(
+        &mut self,
+        config: &CompilerConfig,
+        event_kind: &SourceEventKind,
+    ) -> Result<(), BatchCompileError> {
+        match event_kind {
+            SourceEventKind::CreateOrModify(_) => {}
```

Comment:
this isn't quite right. We should read all the files in the folder. If someone moves a folder from outside of the component directory into the component directory, and it contains iso literals, then these would be unnoticed until we restart the compiler